compiler/rustc_typeck/src/check/inherited.rs

```diff
@@ -124,20 +124,20 @@ impl<'tcx> Inherited<'_, 'tcx> {
 }
 
 impl<'tcx> InheritedBuilder<'tcx> {
-    pub fn enter<F, R>(&mut self, f: F) -> R
+    pub fn enter<F, R>(mut self, f: F) -> R
     where
-        F: for<'a> FnOnce(Inherited<'a, 'tcx>) -> R,
+        F: for<'a> FnOnce(Inherited<'tcx>) -> R,
```

Comment:
`'a` now doesn't exist :grin: though it might be nice to change this function to `&'a FnOnce(&'a Inherited<'tcx>) -> R` so that you still can't use the `Inherited` outside of the closure.

Reply:
Removed `for<'a>` and added a reference to the arg.